databricks/koala/generic.py

```diff
@@ -46,6 +46,40 @@ def compute(self):
         """Alias of `toPandas()` to mimic dask for easily porting tests."""
         return self.toPandas()
 
+    @derived_from(pd.DataFrame)
+    def fillna(self, value=None, axis=None, inplace=False):
+        """Fill NA/NaN values.
+
+        :param value: scalar, dict
+                    Value to use to fill holes. alternately a dict of values
+                    specifying which value to use for each column.
+                    Dataframe and Series are not supported.
+        :param axis: {0 or `index`}
+                    1 and `columns` are not supported.
+        :param inplace: boolean, default False
+                    Fill in place (do not create a new object)
+        :return: :class:`Series` or :class:`DataFrame`
+        """
+        df = self.copy()
```

Comment:
You should not do a copy when you are not doing it in place, all the columns that you may have declared elswhere will not be anchored to the new dataframe.

Reply:
I agree with you.